architecture/2025-01-serverless-feature-spec.md

```diff
@@ -0,0 +1,279 @@
+
+### Scenario 3: User interfaces for serverless--Radius API, CLI, Dashboard
+<!-- One or two sentence summary -->
+Enable deployment and management of serverless compute resources via the existing Radius API and CLI commands. Serverless resources that are modeled in Radius should be available in the App Graph and Dashboard for visualization and management.
+
+## Key dependencies and risks
+<!-- What dependencies must we take in order to enable this scenario? -->
+<!-- What other risks are you aware of that need to be mitigated. If you have a mitigation in mind, summarize here. -->
+<!-- **Dependency Name** – summary of dependency.  Issues/concerns/risks with this dependency -->
+<!-- **Risk Name** – summary of risk.  Mitigation plan if known. If it is not yet known, no problem. -->
+**Dependency: orchestrator and API for the underlying serverless platform** - For each additional serverless platform that Radius supports, we have a dependency on the underlying orchestrator for that platform. For example, for Azure Container Instances, we depend on the Azure API. For AWS Elastic Container Service, we depend on the AWS API. There is a risk that the underlying orchestrator may not provide the necessary APIs to support the Radius model, the APIs may not be stable or reliable, or the APIs may not be publicly available. This will need to be investigated as a part of the implementation.
+
+**Risk: platform specific features that cannot be implemented in Radius** - There might be platform-specific features that cannot (or should not) be implemented in Radius. For example, Kubernetes has features for taints and tolerations that are not common across compute platforms and thus should not be implemented in Radius. This risk can be mitigated by providing mechanisms to punch-through the Radius abstraction and use platform-specific features directly, like the [Kubernetes customization options](https://docs.radapp.io/guides/author-apps/containers/overview/#kubernetes) currently available in Radius.
+
+**Risk: Differing platforms between Radius control plane and application runtime** - There is a need to deploy and maintain a Kubernetes cluster to host the Radius control plane is needed even if the application is completely serverless and this might not work for some customers (e.g. those who absolutely do not want to use Kubernetes). For now we will accept this risk and consider alternative hosting platforms for the Radius control plane to be out of scope.
```

Comment:
Why can't we run Radius on just Docker? I believe Dapr does that with `dapr init` vs `dapr init -k`.

Reply:
It requires a lot of code changes but would that be a good investment? 🧐

Reply:
Do you mean running Radius on Docker within an ACI compute cluster?

Reply:
I mean running Radius on containers (https://docs.dapr.io/getting-started/install-dapr-selfhost/) so that the user wouldn't need a Kubernetes cluster to do a deployment of Serverless resources.

Reply:
ah gotcha, this is a different feature that I think should be tackled when we design and implement https://github.com/orgs/radius-project/projects/8/views/1?pane=issue&itemId=55074612&issue=radius-project%7Croadmap%7C42